private/model/api/shape.go

```diff
@@ -97,7 +97,7 @@ type Shape struct {
 
 	OutputEventStreamAPI *EventStreamAPI
 	EventStream          *EventStream
-	EventFor             []*EventStream `json:"-"`
+	EventFor             map[string]*EventStream `json:"-"`
```

Comment:
Don't think the change will correctly sort the eventStream entries (as range over a map in Go isn't ordered). 

The generation uses 
```
{{ range $_, $eventStream := $.EventFor }}
	// The {{ $.ShapeName }} is and event in the {{ $eventStream.Name }} group of events.
	func (s *{{ $.ShapeName }}) event{{ $eventStream.Name }}() {}
{{ end }}

```

Reply:
This is a special case. Range in template is ordered on key.

Reply:
https://pkg.go.dev/text/template. Covers the `range` helper.